pkg/fanal/analyzer/language/java/jar/jar.go

```diff
@@ -92,6 +92,10 @@ func (a *javaLibraryAnalyzer) PostAnalyze(ctx context.Context, input analyzer.Po
 }
 
 func (a *javaLibraryAnalyzer) Required(filePath string, _ os.FileInfo) bool {
+	// Bitnami images have SBOMs inside, so there is no need to analyze JAR files.
+	if strings.HasPrefix(filePath, "opt/bitnami") {
```

Comment:
I'm not too fond of this approach. @DmitriyLewen Any ideas?

Reply:
What if we will add flag (folder names) insted of this hardcode? 
Flag will show that we need to get packages from sbom instead of language files for those folders.
For `fs,` image`, etc..  `opt/bitnami` will be default value. 

Therefore, users can use this flag for other folders.

Reply:
In the case of Bitnami, SBOM and executable files are located under `opt/bitnami`, but I'm not sure about other cases. I'm afraid the flag is very Bitnami specific.

Reply:
what if we parse both files (I mean jar + sbom) and add logic to exclude duplicates?

Reply:
Yeah, that is what I was thinking first. But for example, JAR scanning is quite expensive (download trivy-java-db, etc.). If there is SBOM, we should skip analyzing.

Reply:
Interesting, but how? `io.Reader` for image layer tarballs is read only once. We cannot iterate twice.
https://github.com/aquasecurity/trivy/blob/48b2e15c2383241963f7bf749d7e4416c33bd4f7/pkg/fanal/walker/tar.go#L40-L94

Reply:
most expensive language packages use post-analyzers ( or we can move them to post-analyzer).
We will add check for filePath from sbom into start of `PostAnalyze` func.
e.g. for jar:
we will add check before downloading trivy-java-db
if bom contains all jar files - we will not download java-db

Correct me, if i am wrong

Reply:
It was impossible since a single SBOM includes several files, and Trivy didn't store those file paths. But it may be possible now after [this change](https://github.com/aquasecurity/go-dep-parser/pull/202/files#diff-8e49337a1417a39a62ebf427e84fb2faa566c96e37ae627ac484c05dbe65c483R15). Trivy has file paths of nested dependencies. I'll give it a shot. Thanks!

Reply:
Added in https://github.com/aquasecurity/trivy/pull/4210/commits/db18245b16dbc1d6bd77445c79674f7628d34f96